### PR TITLE
1001: Fix scrolling and auto domain issue

### PR DESCRIPTION
**Pull request by zBritva**
PR to fix issues reported in https://github.com/microsoft/charticulator/issues/1001, https://github.com/microsoft/charticulator/issues/1002

### CODE REVIEW

File: src/app/stores/app_store.ts [hunk 1]
```diff
@@ -1996,30 +1996,56 @@ export class AppStore extends BaseStore {
...
+                }
+                if (dataBinding.autoDomainMax) {
+                  dataBinding.dataDomainMax = scale.domainMax;
+                } else {
+                  dataBinding.dataDomainMax = options.domainMax;
+                }
               } else {
-                dataBinding.domainMax = options.domainMax;
+                if (dataBinding.autoDomainMin) {
+                  dataBinding.domainMin = scale.domainMin;
```
[the remark this answers is not in the document]
Comment (zBritva) [reply src/app/stores/app_store.ts:2012]: `autoDomainMin` alllows to sync min value of axis with value from new dataset, `domainMin` is the value itself.
There are correspond variables for max too.

[the remark this answers is not in the document]
Comment (zBritva) [reply src/app/stores/app_store.ts:2012]: `domainMax` and `domainMin` define the range of values for the axis (For example values from 5 to 15). `allowScrolling` allows shifting the range up/down (to shift from 5-15 to 25-35)

File: src/core/prototypes/plot_segments/virtualScroll.tsx [hunk 2]
```diff
@@ -82,31 +84,22 @@ export const VirtualScrollBar: React.FC<VirtualScrollBarPropertes> = ({
...
   const [handlePositionX, handlePositionY] = React.useMemo(
     () => mapPositionToCoordinates(position),
     [position, mapPositionToCoordinates]
   );
 
   let handlerWidth = 0;
   let handlerHeight = 0;
-  let buttonsWidth = 0;
-  let buttonsHeight = 0;
+  const buttonsWidth = handlerBarWidth;
```
Comment (MatthLee) [inline src/core/prototypes/plot_segments/virtualScroll.tsx:94]: nit: typo. handle**r**BarWidth should be handleBarWidth (remove "r")

File: src/core/prototypes/plot_segments/virtualScroll.tsx [hunk 3]
```diff
@@ -145,20 +143,27 @@ export const VirtualScrollBar: React.FC<VirtualScrollBarPropertes> = ({
       if (newPosition > 100) {
         newPosition = 100;
       }
-      if (newPosition - widthPerBarPercent * 100 < 0) {
-        newPosition = 0;
+      if (dataType == AxisDataBindingType.Categorical) {
+        if (newPosition - widthPerBarPercent * 100 < 0) {
```
Comment (MatthLee) [inline src/core/prototypes/plot_segments/virtualScroll.tsx:147]: nit: if the variable needs to be multiplied by 100, it isn't a percent, it's a ratio. consider variable name change.

**Review by MatthLee — APPROVED** [review]

**Pull request closed without merging**